### PR TITLE
package app + support directories

### PR DESCRIPTION
**Pull request by MaximeHeckel**
I'll rebase this once all the existing PR's are merged
This cleans up a bit `main.go` by putting most of the functions in packages
@kristiehoward feel free to comment on the naming and my logic to split the code 

I also quickly added the support for directories: 

```
❯ go run main.go -d samples
type ObjectMeta {
  name?: string,
  generateName?: string,
  namespace?: string,
  selfLink?: string,
  uid?: types.UID,
  resourceVersion?: string,
  generation?: number,
  creationTimestamp?: metav1.Time,
  deletionTimestamp?: metav1.Time,
  deletionGracePeriodSeconds?: number,
  labels?: {[string]: string},
  annotations?: {[string]: string},
  ownerReferences?: Array<metav1.OwnerReference>,
  initializers?: metav1.Initializers,
  finalizers?: Array<string>,
  clusterName?: string,
}

type Volume {
  name: string,
}

type VolumeSource {
}

type UpdateProductRequest {
  name: string,
  publisher_id: string,
  created_at: string,
  optional_time?: string,
  nullable_time: ?string,
  short_description: string,
  full_description?: string,
  source: string,
  popularity: number,
  categories: Array<Category>,
  links: Array<URLInfo>,
  screenshots: Array<Array<URLInfo>>,
  logo_url: {[string]: string},
  crazy_map: {[AliasType]: boolean},
  crazy_map_2: {[boolean]: AliasType},
  map_alias_type: MapAliasType,
  is_offline?: boolean,
  array_alias: ArrayAliasType,
  array_pointer_alias: ArrayPointerAliasType,
}

type BoolAlias = boolean;

type AliasType = BoolAlias;

type ArrayAliasType = Array<AliasType>;

type ArrayPointerAliasType = Array<AliasType>;

type MapAliasType = {[boolean]: AliasType};

type URLInfo {
  url: string,
  label: string,
}

type Category {
  name: string,
  label: string,
}

type OperatingSystem {
  name: string,
  label: string,
}
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -34,185 +24,67 @@ var (
...
+
+	// Handle directory
+	if dir != "" {
+		files, err := ioutil.ReadDir(dir)
+		if err != nil {
+			return err
+		}
+
+		for _, f := range files {
+			astNode, err := parse(dir + "/" + f.Name())
```
Comment (kristiehoward) [inline main.go]: can we move the `inspect` into the `parse` fn and maybe rename to `handleFile` or something like that?

then we could do something more like:

```
for _, f := range files {
    err := handleFile(dir + "/" + f.Name())
    if err != nil {
        return err
    }
}

// later in place of lines 85 -92
return handleFile(file)

File: utils/utils_test.go [hunk 2]
```diff
@@ -1,8 +1,10 @@
-package main
+package utils
```
Comment (kristiehoward) [inline utils/utils_test.go]: apparently we arent supposed to name our packages "util"? https://rakyll.org/style-packages/#package-naming

**Review by kristiehoward — APPROVED** [review]
Approved, left some minor comments - nice!

**Comment by kristiehoward** [comment]
You can rebase now!

**Comment by MaximeHeckel** [comment]
Changes applied and rebased!
Feel free to merge @kristiehoward

File: main.go [hunk 3]
```diff
@@ -34,195 +24,72 @@ var (
...
+			err := handleFile(dir + "/" + f.Name())
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	// Handle file
+	err := handleFile(file)
```
Comment (kristiehoward) [inline main.go]: This could be `return handleFile(file)` since we're returning either the err or nil

**Review by kristiehoward — APPROVED** [review]
left a super small comment but this looks great!

File: main.go (hunk 3 above)
Comment (MaximeHeckel) [reply main.go]: 👍

**Comment by MaximeHeckel** [comment]
Just addressed your comment, let's merge this!

**Pull request merged**